tt_metal/programming_examples/matmul_single_core/matmul_single_core.cpp

```diff
@@ -257,96 +198,39 @@ int main(int argc, char **argv) {
-            }
-        }
-
-        vector<uint32_t> result_vec_untilized = pack_bfloat16_vec_into_uint32_vec(untilize(unpack_uint32_vec_into_bfloat16_vec(result_vec), M, N));
-        cout << "----metal_untilized--" << endl;
-        cout << result_vec.size() << endl;
-        for (int i = 0; i < result_vec.size(); i++) {
-            std::pair<bfloat16, bfloat16> as = unpack_two_bfloat16_from_uint32(result_vec_untilized.at(i));
-            float a1 = as.first.to_float();
-            float a2 = as.second.to_float();
-            if (i % 128 == 0){
-                cout << "-- " << i << " -- " << a1<< "  " << a2  << "---" << result_vec_untilized.at(i) << endl;
-            }
-        }
-
-        /* Golden Matmul running on CPU (Float)*/
-        vector<uint32_t> golden_vec;
-        golden_matmul(unpack_uint32_vec_into_bfloat16_vec(src0_vec), unpack_uint32_vec_into_bfloat16_vec(src1_vec), golden_vec, M, N, K, B);
-        vector<uint32_t>  golden_vec_tilized = pack_bfloat16_vec_into_uint32_vec(tilize(unpack_uint32_vec_into_bfloat16_vec(golden_vec), M, N));
-
-        cout << "----golden--" << endl;
-        cout << golden_vec.size() << endl;
-        for (int i = 0; i < golden_vec.size(); i++) {
-            std::pair<bfloat16, bfloat16> as = unpack_two_bfloat16_from_uint32(golden_vec.at(i));
-            float a1 = as.first.to_float();
-            float a2 = as.second.to_float();
-            if (i % 128 == 0){
-                cout << "-- " << i << " -- " << a1 << "  " << a2 << "---" << golden_vec.at(i) << endl;
+        vector<bfloat16> result_vec(dram_buffer_C_size/sizeof(bfloat16));
+        matmul_single_core(src0_vec, src1_vec, result_vec, false, M, N, K, B, device);
+        untilize(result_vec, M, N);
+
+        cout << "---Golden vs Metalium---" << endl;
+        cout << "Output vec size= " << result_vec.size() << endl;
+        bool print_vec_val = 0;
+        if (print_vec_val) {
+            for (int i = 0; i < result_vec.size(); i++) {
+                float a1 = golden_vec.at(i).to_float();
+                float a2 = result_vec.at(i).to_float();
+                if (i % 128 == 0){
+                    cout << "-- " << i << " -- " << a1 << " <--> " << a2 << endl;
+                }
             }
         }
 
-        /* Comparison: Golden vs. METAL Matmul*/
-        constexpr float abs_tolerance = 0.01f;
-        constexpr float rel_tolerance = 0.001f;
-        std::function<bool(const float, const float)> comparison_function = [](const float a, const float b) {
-            return is_close(a, b, rel_tolerance, abs_tolerance);
-        };
-
-        float pearson = packed_uint32_t_vector_pcc_v2(golden_vec_tilized, result_vec);
-        cout << "PCC_v2= " << pearson << endl;
-
+        float pearson = check_bfloat16_vector_pcc(golden_vec, result_vec);
+        cout << "PCC= " << pearson << endl;
         TT_FATAL(pearson > 0.97, "PCC not high");
```

Comment:
Instead of PCC not high, maybe log "PCC not high enough. Result PCC: ..., Expected PCC: ..."

Reply:
done